adal/src/Microsoft.IdentityModel.Clients.ActiveDirectory/Internal/Helpers/EncodingHelper.cs

```diff
@@ -53,8 +55,31 @@ public static string UrlEncode(string message)
                 return message;
             }
 
-            message = Uri.EscapeDataString(message);
-            message = message.Replace("%20", "+");
+            if (message.Length < MaxUrlEncodingSize)
+            {
+                message = Uri.EscapeDataString(message);
+                message = message.Replace("%20", "+");
+            }
+            else
+            {
+                StringBuilder sb = new StringBuilder();
```

Comment:
Please add unit tests for this change, it is quite complex.

Reply:
@bgavrilMS Added unit test